docs/source/unsloth_integration.md

```diff
@@ -1,7 +1,197 @@
-# Unsloth Integration
+# Unsloth Integration Guide
 
-<Tip warning={true}>
+Unsloth is a lightweight library designed to accelerate and optimize the training of large language models (LLMs) while reducing memory usage. This guide provides step-by-step instructions for integrating Unsloth into your existing workflows.
 
-Section under construction. Feel free to contribute!
+---
 
-</Tip>
\ No newline at end of file
+## Installation Instructions
+
+### Stable Releases
+For stable releases, use the following command:
+```bash
+pip install unsloth
+```
+
+### Recommended Installation
+For most installations, we recommend using:
+```bash
+pip install "unsloth[colab-new] @ git+https://github.com/unslothai/unsloth.git"
+```
+
+---
+
+## System Requirements
+
+- **Operating System**: Works on Linux and Windows via WSL.
+- **GPU Support**: Supports NVIDIA GPUs since 2018+. Minimum CUDA Capability 7.0 (V100, T4, Titan V, RTX 20, 30, 40x, A100, H100, L40, etc.). Check your GPU! GTX 1070, 1080 works, but is slow.
+- **Dependencies**: Your device must have `xformers`, `torch`, `BitsandBytes`, and `triton` support.
+- **Disk Space**: Ensure you have sufficient disk space to train and save your model.
+
+**Note**: Unsloth only works if you have an NVIDIA GPU.
+
+---
+
+## Fine-tuning VRAM Requirements
+
+How much GPU memory do you need for LLM fine-tuning using Unsloth?  
+Check the table below for VRAM requirements sorted by model parameters and fine-tuning method.  
+- **QLoRA** uses 4-bit precision.  
+- **LoRA** uses 16-bit precision.
+
+| Model Parameters | QLoRA (4-bit) VRAM | LoRA (16-bit) VRAM |
+|------------------|-------------------|--------------------|
+| 3B               | 2 GB              | 7 GB               |
+| 7B               | 4.5 GB            | 16 GB              |
+| 8B               | 5 GB              | 19 GB              |
+| 9B               | 5.5 GB            | 21 GB              |
+| 11B              | 6.5 GB            | 26 GB              |
+| 14B              | 8.5 GB            | 33 GB              |
+| 27B              | 16 GB             | 64 GB              |
+| 32B              | 19 GB             | 76 GB              |
+| 40B              | 24 GB             | 96 GB              |
+| 70B              | 41 GB             | 164 GB             |
+| 81B              | 48 GB             | 192 GB             |
+| 405B             | 237 GB            | 950 GB             |
+
+## Training Setup
+```python
+from transformers import TrainingArguments
+from trl import SFTTrainer  # Fixed trainer import 
```

Comment:
wdym by "fixed trainer import"? 
also, use `SFTConfig` instead of `TrainingArguments` here